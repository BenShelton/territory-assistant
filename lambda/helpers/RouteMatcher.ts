import { APIGatewayEvent } from 'aws-lambda'

type APIMethod = 'GET' | 'POST' | 'PUT' | 'DELETE'

export class RouteMatcher {
  private _paramRegex = /:(.*?)(?=(\/|$))/
  private _event: APIGatewayEvent
  private _pathParams: Record<string, string> = {}

  constructor (event: APIGatewayEvent) {
    this._event = event
  }

  get pathParams (): Record<string, string> {
    return this._pathParams
  }

  private _getMatches (path: string, regex: RegExp): string[] {
    const matches = path.match(regex)
    if (!matches) return []
    return matches.slice(1)
  }

  testRoute (method: APIMethod, path: string): boolean {
    if (this._event.httpMethod !== method) return false
    const stringParamRegex = new RegExp(path.replace(this._paramRegex, '(\\w+)') + '/?$')
    if (!stringParamRegex.test(this._event.path)) return false
    const paramNames = this._getMatches(path, this._paramRegex)
    const paramValues = this._getMatches(this._event.path, stringParamRegex)
    this._pathParams = paramNames.reduce((acc: Record<string, string>, p, i) => {
      if (p) acc[p] = paramValues[i]
      return acc
    }, {})
    return true
  }
}
