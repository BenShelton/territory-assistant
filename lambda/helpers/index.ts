import { APIGatewayEvent } from 'aws-lambda'

type APIMethod = 'GET' | 'POST'
type RouteMatcher = (method: APIMethod, path: string) => boolean
interface IResult {
  statusCode: number
  body: string
}

export function generateRouteMatcher (event: APIGatewayEvent): RouteMatcher {
  return function routeMatcher (method: APIMethod, path: string) {
    if (event.httpMethod !== method) return false
    return new RegExp(path + '/?$').test(event.path)
  }
}

export function success<T> (data: T): IResult {
  return { statusCode: 200, body: JSON.stringify(data) }
}

export function badRequest (message: string): IResult {
  return { statusCode: 400, body: message }
}

export function notFound (): IResult {
  return { statusCode: 404, body: 'Unknown Route' }
}
