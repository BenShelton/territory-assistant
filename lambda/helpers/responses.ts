interface IResult {
  statusCode: number
  body: string
}

export function success<T> (data: T): IResult {
  return { statusCode: 200, body: JSON.stringify(data) }
}

export function badRequest (message: string): IResult {
  return { statusCode: 400, body: message }
}

export function unauthorized (): IResult {
  return { statusCode: 401, body: 'Unauthorized' }
}

export function notFound (): IResult {
  return { statusCode: 404, body: 'Unknown Route' }
}
