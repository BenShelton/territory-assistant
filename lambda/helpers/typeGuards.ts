import { IPoint } from 'types'

export function isObject (obj: unknown): obj is Record<string, unknown> {
  return typeof obj === 'object' && !Array.isArray(obj) && !!obj
}

export function isPointList (obj: unknown): obj is IPoint[] {
  if (!Array.isArray(obj)) return false
  const basePoint: IPoint = { lat: 0, lng: 0 }
  return Object.entries(basePoint)
    .every(([k, v]) => {
      return obj.every(p => {
        if (!isObject(p)) return false
        return typeof p[k] === typeof v
      })
    })
}
