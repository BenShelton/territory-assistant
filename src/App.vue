<template>
  <v-app>
    <AppBar />
    <NavigationDrawer />
    <v-main>
      <router-view />
    </v-main>
    <Notification />
  </v-app>
</template>

<script lang="ts">
import Vue from 'vue'

import AppBar from '@/components/AppBar.vue'
import NavigationDrawer from '@/components/NavigationDrawer.vue'
import Notification from '@/components/Notification.vue'

export default Vue.extend({
  name: 'App',

  components: {
    AppBar,
    NavigationDrawer,
    Notification
  }
})
</script>
