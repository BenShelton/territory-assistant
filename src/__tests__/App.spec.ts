import { createWrapper } from 'UnitTest/helpers'

import App from '../App.vue'

describe('App', () => {
  it('should have the name "App"', () => {
    const wrapper = createWrapper(App, { store: true })
    expect(wrapper.name()).toBe('App')
  })

  it('should match the snapshot', () => {
    const wrapper = createWrapper(App, { store: true })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
