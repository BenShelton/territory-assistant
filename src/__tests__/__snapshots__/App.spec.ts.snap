// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div data-app="true" class="v-application v-application--is-ltr theme--light" id="app">
  <div class="v-application--wrap">
    <header class="v-sheet v-sheet--tile theme--light v-toolbar v-app-bar v-app-bar--fixed" style="height: 64px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;">
      <div class="v-toolbar__content" style="height: 64px;">
        <!---->
        <div class="v-toolbar__title headline"><span>Territory Assistant</span></div>
      </div>
    </header>
    <nav class="v-navigation-drawer v-navigation-drawer--close v-navigation-drawer--fixed theme--light" style="height: 100vh; top: 0px; max-height: calc(100% - 0px); transform: translateX(-100%); width: 256px;">
      <div class="v-navigation-drawer__content">
        <header class="v-sheet v-sheet--tile theme--light v-toolbar" style="height: 64px;">
          <div class="v-toolbar__content" style="height: 64px;"><button type="button" class="v-btn v-btn--flat v-btn--icon v-btn--round theme--light v-size--default"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-close theme--light"></i></span></button>
            <div class="v-toolbar__title">Navigation</div>
          </div>
        </header>
        <div class="v-list v-sheet v-sheet--tile theme--light v-list--nav v-list--subheader">
          <div class="v-subheader theme--light">Overview</div> <a href="/" class="v-list-item--active v-list-item v-list-item--link theme--light" tabindex="0">
            <div class="v-list-item__icon"><i aria-hidden="true" class="v-icon notranslate mdi mdi-home theme--light"></i></div>
            <div class="v-list-item__title">Home</div>
          </a>
        </div>
        <div class="v-list v-sheet v-sheet--tile theme--light v-list--nav v-list--subheader">
          <div class="v-subheader theme--light">Editors</div> <a href="/territory" class="v-list-item v-list-item--link theme--light" tabindex="0">
            <div class="v-list-item__icon"><i aria-hidden="true" class="v-icon notranslate mdi mdi-map theme--light"></i></div>
            <div class="v-list-item__title">Territory</div>
          </a><a href="/image" class="v-list-item v-list-item--link theme--light" tabindex="0">
            <div class="v-list-item__icon"><i aria-hidden="true" class="v-icon notranslate mdi mdi-image theme--light"></i></div>
            <div class="v-list-item__title">Image</div>
          </a><a href="/maps" class="v-list-item v-list-item--link theme--light" tabindex="0">
            <div class="v-list-item__icon"><i aria-hidden="true" class="v-icon notranslate mdi mdi-map-search theme--light"></i></div>
            <div class="v-list-item__title">Maps</div>
          </a><a href="/info" class="v-list-item v-list-item--link theme--light" tabindex="0">
            <div class="v-list-item__icon"><i aria-hidden="true" class="v-icon notranslate mdi mdi-map-marker theme--light"></i></div>
            <div class="v-list-item__title">Info</div>
          </a>
        </div>
      </div>
      <div class="v-navigation-drawer__append">
        <div class="v-list v-sheet v-sheet--tile theme--light v-list--nav">
          <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light"> <a href="/settings" class="v-list-item v-list-item--link theme--light" tabindex="0">
            <div class="v-list-item__icon"><i aria-hidden="true" class="v-icon notranslate mdi mdi-settings theme--light"></i></div>
            <div class="v-list-item__title">Settings</div>
          </a>
          <div tabindex="0" class="v-list-item v-list-item--link theme--light">
            <div class="v-list-item__icon"><i aria-hidden="true" class="v-icon notranslate mdi mdi-logout-variant theme--light"></i></div>
            <div class="v-list-item__title">Logout</div>
          </div>
        </div>
      </div>
      <div class="v-navigation-drawer__border"></div>
    </nav>
    <main class="v-content" style="padding: 64px 0px 0px 0px;">
      <div class="v-content__wrap"></div>
    </main>
    <div class="notifications-wrapper"><span></span></div>
  </div>
</div>
`;
